compiler/rustc_hir_typeck/src/generator_interior/mod.rs

```diff
@@ -92,6 +94,13 @@ impl<'a, 'tcx> InteriorVisitor<'a, 'tcx> {
                 Some(YieldData { span: DUMMY_SP, expr_and_pat_count: 0, source: self.kind.into() })
             });
 
+        // If this is the `&mut Context<'_>` async resume argument, or we are
+        // just visiting a `_task_context = yield ()` expression from async
+        // lowering, we do *not* consider this type to be live across yields.
+        if Some(hir_id) == self.task_context_hid || self.in_await_yield {
```

Comment:
Can you explain why we need _both_ of these checks? Shouldn't we always be accessing this context type through the same local variable? Could you drop `in_await_yield` and maybe match on `expr` instead? 

I'm thinking that it should be a `ExprKind::Path` or something... you could see that it resolved to the argument we're expecting it to be.

Reply:
I tried matching just on the `ExprKind::Path` of the lhs of the assignment. The problem is also the `yield` itself, more specifically the `()` going into it, and the `&mut Context<'_>` coming out of it.

Fun fact, by skipping the whole expression, it also avoids putting a `()` into the generator witness in some cases.